build/nested-cluster/start.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+wait_for_docker ()
+{
+  # Wait for docker.
+  until docker version; do sleep 1 ;done
+}
+
+start_kubelet ()
+{
+  # Start the kubelet.
+  mkdir -p /etc/kubernetes/manifests
+  mkdir -p /etc/srv/kubernetes
+
+  # Change the kubelet to not fail with swap on.
+  cat > /etc/systemd/system/kubelet.service.d/kubeadm-20.conf << EOM
+[Service]
+Environment="KUBELET_EXTRA_ARGS=--fail-swap-on=false"
+EOM
+#  sed -i 's/ExecStart=\/usr\/bin\/kubelet/ExecStart=\/usr\/bin\/kubelet --fail-swap-on=false/' /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+  systemctl enable kubelet
+  systemctl start kubelet
+}
+
+start_worker ()
+{
+  wait_for_docker
+  start_kubelet
+
+  # Load docker images
+  docker load -i /kube-proxy-amd64.tar
+
+  # Start kubeadm.
+  /usr/bin/kubeadm join --token=abcdef.abcdefghijklmnop --discovery-token-unsafe-skip-ca-verification=true --ignore-preflight-errors=all 172.18.0.2:6443 2>&1
+}
+
+start_master ()
+{
+  wait_for_docker
+  mount --make-rshared /etc/kubernetes
+  start_kubelet
+
+  # Load the docker images
+  docker load -i /cloud-controller-manager-amd64.tar
```

Comment:
Once CI is working, I'll spend some time making this faster. Since this is the critical path for startup, it's almost certain I'll change how it works.